x/dclupgrade/keeper/grpc_query_approved_upgrade_test.go

```diff
@@ -0,0 +1,126 @@
+package keeper_test
+
+import (
+	"strconv"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	keepertest "github.com/zigbee-alliance/distributed-compliance-ledger/testutil/keeper"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/testutil/nullify"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/dclupgrade/types"
+)
+
+// Prevent strconv unused error.
+var _ = strconv.IntSize
+
+func TestApprovedUpgradeQuerySingle(t *testing.T) {
+	keeper, ctx := keepertest.DclupgradeKeeper(t, nil, nil)
+	wctx := sdk.WrapSDKContext(ctx)
+	msgs := createNApprovedUpgrade(keeper, ctx, 2)
+	for _, tc := range []struct {
+		desc     string
+		request  *types.QueryGetApprovedUpgradeRequest
+		response *types.QueryGetApprovedUpgradeResponse
+		err      error
+	}{
+		{
+			desc: "First",
+			request: &types.QueryGetApprovedUpgradeRequest{
+				Name: msgs[0].Plan.Name,
+			},
+			response: &types.QueryGetApprovedUpgradeResponse{ApprovedUpgrade: msgs[0]},
+		},
+		{
+			desc: "Second",
+			request: &types.QueryGetApprovedUpgradeRequest{
+				Name: msgs[1].Plan.Name,
+			},
+			response: &types.QueryGetApprovedUpgradeResponse{ApprovedUpgrade: msgs[1]},
+		},
+		{
+			desc: "KeyNotFound",
+			request: &types.QueryGetApprovedUpgradeRequest{
+				Name: strconv.Itoa(100000),
+			},
+			err: status.Error(codes.InvalidArgument, "not found"),
```

Comment:
This should be corrected correspondingly to the corrections in `Keeper.ApprovedUpgrade` method.

Reply:
fixed